interop-testing/src/main/java/io/grpc/testing/integration/GrpclbFallbackTestClient.java

```diff
@@ -0,0 +1,296 @@
+    int exitCode = process.waitFor();
+    logger.info("Shell rommand exit code: " + exitCode);
+    assertEquals(0, exitCode);
+  }
+
+  private enum RpcMode {
+    FailFast,
+    WaitForReady,
+  }
+
+  private GrpclbRouteType doRpcAndGetPath(
+      TestServiceGrpc.TestServiceBlockingStub blockingStub,
+      int deadlineSeconds,
+      RpcMode rpcMode) {
+    logger.info("doRpcAndGetPath deadlineSeconds: " + deadlineSeconds + " rpcMode: "
+        + rpcMode);
+    final SimpleRequest request = SimpleRequest.newBuilder()
+        .setFillGrpclbRouteType(true)
+        .build();
+    GrpclbRouteType result = GrpclbRouteType.GRPCLB_ROUTE_TYPE_UNKNOWN;
+    try {
+      SimpleResponse response = null;
+      if (rpcMode == RpcMode.WaitForReady) {
+        response = blockingStub
+          .withDeadlineAfter(deadlineSeconds, TimeUnit.SECONDS)
+          .withWaitForReady()
+          .unaryCall(request);
+      } else {
+        response = blockingStub
+          .withDeadlineAfter(deadlineSeconds, TimeUnit.SECONDS)
+          .unaryCall(request);
+      }
+      result = response.getGrpclbRouteType();
+    } catch (StatusRuntimeException ex) {
+      logger.warning("doRpcAndGetPath failed. Status: " + ex);
+    }
+    logger.info("doRpcAndGetPath. GrpclbRouteType result: " + result);
+    if (result != GrpclbRouteType.GRPCLB_ROUTE_TYPE_FALLBACK
+        && result != GrpclbRouteType.GRPCLB_ROUTE_TYPE_BACKEND) {
+      throw new AssertionError("Received invalid LB route type. This suggests "
+          + "that the server hasn't implemented this test correctly.");
+    }
+    return result;
+  }
+
+  private void doFallbackBeforeStartupTest(
+      String breakLbAndBackendConnsCmd, int deadlineSeconds) throws Exception {
+    runShellCmd(breakLbAndBackendConnsCmd);
+    for (int i = 0; i < 30; i++) {
+      GrpclbRouteType grpclbRouteType = doRpcAndGetPath(
+          blockingStub, deadlineSeconds, RpcMode.FailFast);
+      assertEquals(grpclbRouteType, GrpclbRouteType.GRPCLB_ROUTE_TYPE_FALLBACK);
+      Thread.sleep(1000);
+    }
+  }
+
+  private void runFastFallbackBeforeStartup() throws Exception {
+    doFallbackBeforeStartupTest(unrouteLbAndBackendAddrsCmd, 9);
+  }
+
+  private void runSlowFallbackBeforeStartup() throws Exception {
+    doFallbackBeforeStartupTest(blackholeLbAndBackendAddrsCmd, 20);
+  }
+
+  private void doFallbackAfterStartupTest(
+      String breakLbAndBackendConnsCmd) throws Exception {
+    assertEquals(
+        doRpcAndGetPath(blockingStub, 20, RpcMode.FailFast),
+        GrpclbRouteType.GRPCLB_ROUTE_TYPE_BACKEND);
+    runShellCmd(breakLbAndBackendConnsCmd);
+    for (int i = 0; i < 40; i++) {
+      GrpclbRouteType grpclbRouteType = doRpcAndGetPath(
+          blockingStub, 1, RpcMode.WaitForReady);
+      if (grpclbRouteType != GrpclbRouteType.GRPCLB_ROUTE_TYPE_BACKEND) {
+        throw new AssertionError("Got grpclb route type backend. Backends are "
```

Comment:
You did _not_ get route type backend. Is the condition wrong or the message?

Reply:
The condition was wrong, fixed.

Thanks for comments, sorry for batch of errors. I pushed a half-done set of changes earlier